BrainPortal/app/controllers/demands_controller.rb

```diff
@@ -0,0 +1,437 @@
+
+
+
+  def new #:nodoc:
+    @demand = Demand.new
+  end
+
+
+
+  def create #:nodoc:
+    @demand = Demand.new(params[:demand])
+    @demand.session_id = request.session_options[:id]
+    @demand.generate_token
+
+    unless can_edit?(@demand)
+      redirect_to '/login'
+      return
+    end
+
+    if ! @demand.save
+      render :action => :new
+      return
+    end
+
+    unless send_confirm_email(@demand)
+      flash[:error] = "It seems some error occured. Email notification was probably not sent.\n"
+    end
+
+    send_admin_notification(@demand)
+
+    sleep 1
+    redirect_to :action => :show, :id => @demand.id
+  end
+
+
+
+  def edit #:nodoc:
+    @demand = Demand.find(params[:id]) rescue nil
+    unless can_edit?(@demand)
+      redirect_to '/login'
+      return
+    end
+    render :action => :new
+  end
+
+
+
+  def update #:nodoc:
+    @demand = Demand.find(params[:id]) rescue nil
+
+    unless can_edit?(@demand)
+      redirect_to '/login'
+      return
+    end
+
+    @demand.update_attributes(params[:demand])
+
+    if ! @demand.save
+      render :action => :new
+      return
+    end
+
+    flash[:notice] = "The account request has been updated."
+
+    sleep 1
+    redirect_to :action => :show, :id => @demand.id
+  end
+
+
+
+  def index #:nodoc:
+    @scope = scope_from_session('demands')
+
+    scope_default_order(@scope, 'country')
+
+    @base_scope       = Demand.where({})
+    @demands          = @scope.apply(@base_scope)
+
+    # Prepare the Pagination object
+    @scope.pagination ||= Scope::Pagination.from_hash({ :per_page => 25 })
+    @current_offset = (@scope.pagination.page - 1) * @scope.pagination.per_page
+
+    scope_to_session(@scope, 'demands')
+
+    respond_to do |format|
+      format.js
+      format.html # index.html.erb
+    end
+
+  end
+
+
+
+  def destroy #:nodoc:
+    @demand = Demand.find(params[:id]) rescue nil
+
+    unless can_edit?(@demand)
+      redirect_to '/login'
+      return
+    end
+
+    @demand.destroy
+    flash[:notice] = "The account request has been deleted."
+
+    redirect_to :action => :index
+  end
+
+
+  # Confirms that a demandee's email address actually belongs to them
+  def confirm #:nodoc:
+    @demand = Demand.find(params[:id]) rescue nil
+    token    = params[:token] || ""
+    if @demand.present? && token.present? && @demand.confirm_token == token
+      @demand.confirmed = true
+      @demand.save
+    else
+      redirect_to '/login'
+    end
+  end
+
+
+  # Administrator action that converts a demand into a user
+  def approve
+    @demand = Demand.find(params[:id]) rescue nil
+
+    unless can_edit?(@demand)
+      flash[:error] = "Could not approve account"
+      redirect_to '/login'
+      return
+    end
+
+    if @demand.login.blank?
+      flash[:error] = "Before approval, a 'login' name must be set."
+      redirect_to :action => :edit, :id => @demand.id
+      return
+    end
+
+    symbolic_result, @info, @exception_trace = approve_one(@demand)
```

Comment:
I created a UserGroup with the same username as my new user, to prevent the user object from being created, and when I clicked approved I got conflicting message:

![screen shot 2016-09-29 at 15 04 29](https://cloud.githubusercontent.com/assets/777588/18968271/5279e200-8656-11e6-8f39-4fb30efb704f.png)